### PR TITLE
Tests for Gauss-Legendre and Gauss-Lobatto rules for N=2,..,100 points.

### LINKED ISSUE #37: Precomputed weights and nodes?
The angular grids are precomputed and stored statically.

I wonder whether it would make sense to also precompute the primitive quadratures, since this could be done in arbitrary precision with e.g. `sympy`.

Storing the weights and nodes for $n$ points costs $2 n$ floating point numbers. This means that storing all rules from $n=1$ to $N$ costs $(N+1)^2 - N -1$. 

| $N$ | Storage cost |
|---|---|
|100|80 kB|
|250|52 kB|
|500|2 MB|
|2500|50 MB|

It might be a good idea to to have this functionality anyway, since one should test the accuracy of the computed nodes and weights by comparison to accurate reference values.

### PR DESCRIPTION
**Pull request by susilehtola**
Closes #37.

The tests are easy to extend to higher numbers of nodes.

The Gauss-Legendre tests fail due to #40. I disabled the Gauss-Lobatto tests until #45 is merged.

### CODE REVIEW

File: test/generate_quadrature_tests.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/* THIS FILE IS AUTOGENERATED BY generate_quadrature_tests.py.
+   DO NOT EDIT THIS FILE, EDIT THE AUTOGENERATOR INSTEAD.
+*/
+
+#include "catch2/catch_all.hpp"
+#include <integratorxx/quadratures/{rule.lower()}.hpp>
+#include <cfloat>
+
+const double x_tolerance = 10*DBL_EPSILON;
+const double w_tolerance = 10*DBL_EPSILON;
```
Comment (wavefunction91) [inline test/generate_quadrature_tests.py:49]:
```suggestion
#include <numeric>

const double x_tolerance = 10*std::numeric_limits<double>::epsilon();
const double w_tolerance = 10*std::numeric_limits<double>::epsilon();
```

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,12 @@ target_link_libraries( quadrature_manipulation PUBLIC Catch2::Catch2WithMain int
 add_executable( 1d_quadratures 1d_quadratures.cxx )
 target_link_libraries( 1d_quadratures PUBLIC Catch2::Catch2WithMain integratorxx )
 
+add_executable( gausslegendre gausslegendre.cxx )
+target_link_libraries( gausslegendre PUBLIC Catch2::Catch2WithMain integratorxx )
+
+#add_executable( gausslobatto gausslobatto.cxx )
+#target_link_libraries( gausslobatto PUBLIC Catch2::Catch2WithMain integratorxx )
```
Comment (wavefunction91) [inline test/CMakeLists.txt:32]: Why is this commented out?

Also, if these are to be run in the unit test system, you must also add appropriate `add_test` lines to register them with `ctest`

**Review by wavefunction91 — CHANGES_REQUESTED** [review]
One tedious point - just to make these files more readable, I would probably have opted to store the reference data in binary files (e.g. HDF5), but I won't impose that for this PR as I agree that it's important to get this kind of testing in place.

File: test/CMakeLists.txt (hunk 2 above)
Comment (susilehtola) [reply test/CMakeLists.txt:32]: Like I commented above, these are commented out since Lobatto quadrature isn't yet in the master branch!

Comment (susilehtola) [reply test/CMakeLists.txt]: Tests added

File: test/generate_quadrature_tests.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        print(f'{fname} already exists, skipping')
+        continue
+
+    out=open(fname,'w')
+    out.write(f'''
+/* THIS FILE IS AUTOGENERATED BY generate_quadrature_tests.py.
+   DO NOT EDIT THIS FILE, EDIT THE AUTOGENERATOR INSTEAD.
+*/
+
+// clang-format off
```
Comment (wavefunction91) [inline test/generate_quadrature_tests.py:60]: Does `clang-format` mess up the header ordering?

**Review by wavefunction91 — APPROVED** [review]
If this is g2g, I'll merge, just let me know.

File: test/generate_quadrature_tests.py (hunk 3 above)
Comment (susilehtola) [reply test/generate_quadrature_tests.py:60]: Sure does

**Comment by susilehtola** [comment]
> If this is g2g, I'll merge, just let me know.

Good to go in my opinion. The stuff for #49 can be added later

**Pull request merged**